src/main/java/xyz/aadev/generitech/common/tileentities/machines/TileEntityFurnace.java

```diff
@@ -162,22 +169,35 @@ public float getTempRate() {
 
     @Override
     public void update() {
-        if (machineActive & !isSmeltPaused) {
+        if (container.getStoredPower()>=powerUsage && machineActive & !isSmeltPaused) {
             if (internalTemp < this.getMaxTemperature()) {
+                System.out.println("MaxTemperature");
                 internalTemp += getTempRate();
+                container.takePower(powerUsage,false);
             }
-        } else {
-            if (canIdle && internalTemp <= getIdleTemp()) {
-                internalTemp = getIdleTemp();
+        }else if (container.getStoredPower()>=powerUsage&&canIdle && internalTemp <= getIdleTemp())
+        {
+            System.out.println("getIdleTemp");
+            internalTemp += getTempRate();
+            container.takePower(powerUsage,false);
+        }else {
+            if (internalTemp <= 0) {
+                internalTemp = 0;
             } else {
-                if (internalTemp <= 0) {
-                    internalTemp = 0;
-                } else {
-                    internalTemp -= getTempRate() / 2;
-                }
+                internalTemp -= getTempRate() / 2;
             }
         }
 
+        BlockPos pos = getPos();
+        World worldIn = getWorld();
+        if (worldIn.isBlockPowered(pos)&&canIdle!=true){
+            canIdle = true;
+        }
+        if (!worldIn.isBlockPowered(pos)&&canIdle!=false){
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1125)